keepalived/check/check_api.c

```diff
@@ -117,6 +120,30 @@ queue_checker(void (*free_func) (void *), void (*dump_func) (void *)
 	}
 }
 
+int
+compare_conn_opts(conn_opts_t *a, conn_opts_t *b)
+{
+	if (a == b)
+		return 0;
+
+	if (!a || !b)
+		goto err;
```

Comment:
Change all `goto err` statements to `return false` in all the compare functions. `goto err` suggests an error, which this is not, it is simple that they do not compare the same.